### ISSUE TITLE
working directory

### ISSUE THREAD

**Issue by fuleshu** [octref/vscode-electron-debug#1]
Hi,
thanks for your good description.
However, I have the problem that I load files from the app directory, but the working directory now is the visual studio code directory. Probably the
"webRoot": "${workspaceRoot}"
was supposed to change the working directory, but it doesn't do it. Any ideas how to set the working directory?
Thanks,
Timo

**Comment by octref** [comment]
Hi Timo. `${workspaceRoot}` is the root directory of the project you are working on.

If your project is like

```
my_project/
my_project/app/
my_project/app/index.html
```

And you are doing `code my_project`.

Change `webRoot` to `${workspaceRoot}/app`.

**Comment by fuleshu** [comment]
Thanks for the fast reply.
Unfortunately this doesn't help. It seems that the `webRoot` is ignored.
I actually use a loki database and load it with:
`var db = new loki('loki_db.json');`
But no matter what I put in the `webRoot` I get:
`Uncaught Error: EPERM: operation not permitted, open 'C:\Program Files (x86)\Microsoft VS Code\loki_db.json'`

For my normal electron launch I use:
`"cwd": "${workspaceRoot}/app/."`
and this works fine.

**Comment by octref** [comment]
Can you paste your `launch.json` and tell me which target you are using?

Also, are you using

```
var db = new loki('loki_db.json');
```

in your renderer or main process?

From what I see in Loki's website, it seems to be a in-memory DB. So you should put it in your renderer process. However it seems you are doing it in your main process.

**Comment by fuleshu** [comment]
My `launch.json` is:

``` json
{
    "version": "0.1.0",
    "configurations": [
        { 
            "name": "Launch Electron App",
            "type": "node",
            "program": "${workspaceRoot}/app/main.js", // important
            "stopOnEntry": false,
            "args": [],
            "cwd": "${workspaceRoot}/app/.",
            // next line is very important
            "runtimeExecutable": "${workspaceRoot}/app/node_modules/electron-prebuilt/dist/electron.exe",
            "runtimeArguments": [],
            "env": { }, 
            "sourceMaps": false
        },
        {
            "name": "Debug",
            "type": "chrome",
            "request": "launch",
            "runtimeExecutable": "${workspaceRoot}/app/node_modules/electron-prebuilt/dist/electron.exe",
            "runtimeArgs": [
                "${workspaceRoot}/app/.",
                "--enable-logging",
                "--remote-debugging-port=9222"
            ],
            "webRoot": "${workspaceRoot}/app/.",
            "sourceMaps": false
        }
    ]
}
```

Sorry, I am still not really sure about what goes in the main or render thread. I put the db functions in a service. The service script is included in the `index.html`
`<script src="./components/service/dbService.js"></script>`
And first called in the controller of the main page. So I guess it's render thread?

**Comment by octref** [comment]
I just tried adding loki to the `electron-quick-start` on my Mac, and it works fine. I pushed the changes to the branch `loki`. Take a look at the `electron-quick-start` folder: https://github.com/octref/vscode-electron-debug/tree/loki/electron-quick-start

Here is the commit:
https://github.com/octref/vscode-electron-debug/commit/ca59c6aa1707ef68d342ef54281df5fcb71edefa

I doubt it's a loki issue with Windows.

**Comment by fuleshu** [comment]
Sorry, I didn't write the complete lokijs loading before. After creating the db you still have to load it.
I took your example and after:
`window.loki_test_db = new loki('loki_db.json')`
I added:

``` javascript
  window.loki_test_db.loadDatabase({}, function (result) 
  {
            console.log('loki loaded');
            console.log(result);
  });
```

Now I get the same exception as in my software:
`Error {message: "Error: ENOENT: no such file or directory, open 'C:…ogram Files (x86)\Microsoft VS Code\loki_db.json'"}`
also
`window.loki_test_db.saveDatabase();`
has the same problem. It tries to save it in the VS directory and throws an exception that it doesn't have the right to write there.
